// zkverifyjs/dist/commonjs/proofTypes/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Plonky2Processor = exports.ProofOfSqlProcessor = exports.UltraPlonkProcessor = exports.Risc0Processor = exports.Groth16Processor = exports.FflonkProcessor = void 0;
var processor_1 = require("./fflonk/processor");
Object.defineProperty(exports, "FflonkProcessor", { enumerable: true, get: function () { return __importDefault(processor_1).default; } });
var processor_2 = require("./groth16/processor");
Object.defineProperty(exports, "Groth16Processor", { enumerable: true, get: function () { return __importDefault(processor_2).default; } });
var processor_3 = require("./risc0/processor");
Object.defineProperty(exports, "Risc0Processor", { enumerable: true, get: function () { return __importDefault(processor_3).default; } });
var processor_4 = require("./ultraplonk/processor");
Object.defineProperty(exports, "UltraPlonkProcessor", { enumerable: true, get: function () { return __importDefault(processor_4).default; } });
var processor_5 = require("./proofofsql/processor");
Object.defineProperty(exports, "ProofOfSqlProcessor", { enumerable: true, get: function () { return __importDefault(processor_5).default; } });
var processor_6 = require("./plonky2/processor");
Object.defineProperty(exports, "Plonky2Processor", { enumerable: true, get: function () { return __importDefault(processor_6).default; } });
// ADD_NEW_PROOF_TYPE
